### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [Schniz/fnm#1284]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I found [this](https://github.com/Schniz/fnm/pull/747) with introducing additional optimization options to the project (including LTO) but it was silently closed. Maybe it's a good time to consider enabling LTO once again ;)

Thank you.

P.S. It's an idea not an actual issue, so maybe Discussions with the "Ideas" section is a better place to discuss such things (but for now Discussions are disabled in this repo)

**Comment by Schniz** [comment]
Closing this — it's been open for a while without activity. If you're still seeing this on a recent version, feel free to reopen.

---

*This message was written by AI. Sorry it's not personal, but it's helping me maintain this project. Feel free to disagree with the statement and reopen the issue if needed.*

**Comment by zamazan4ik** [comment]
@Schniz it's still not fixed - reopen the issue
